src/Downloader.php

```diff
@@ -412,4 +412,11 @@ private function initialize()
             $this->endpoints[$endpoint] = $config;
         }
     }
+
+    private static function generateCacheKey(string $url): string
+    {
+        $url = preg_replace('@^https://api.github.com/repos/symfony/recipes(-contrib)?/contents/@', 'sf-recipes$1/', $url);
+
+        return preg_replace('{[^a-z0-9.]}i', '-', $url);
```

Comment:
to be sure that this bug won't occurs again, you could truncate the string (and adding hash to prevent collisions
```php
$maxLength = ???;
if (strlen($url) > $maxLength) {
  $url = substr($url, 0, $maxLength - 8).'-'.substr(sha1($url), 0, 7);
}

Reply:
let's keep things simple for now at least :)